File: trunk/java/core/src/org/daisy/urakawa/property/channel/AudioChannel.java
package org.daisy.urakawa.property.channel;

import org.daisy.urakawa.exception.MethodParameterIsNullException;
import org.daisy.urakawa.media.IAudioMedia;
import org.daisy.urakawa.media.IMedia;
import org.daisy.urakawa.media.ISequenceMedia;

/**
 *
 */
public class AudioChannel extends Channel
{
    @Override
    public boolean canAccept(IMedia m) throws MethodParameterIsNullException
    {
        if (!super.canAccept(m))
            return false;
        if (m instanceof IAudioMedia)
            return true;
        if (m instanceof ISequenceMedia)
        {
            for (IMedia sm : ((ISequenceMedia) m).getListOfItems())
            {
                if (!(sm instanceof IAudioMedia))
                    return false;
            }
            return true;
        }
        return false;
    }
}

File: branches/1.1_before_major_refactor_2.0/csharp/core/property/channel/AudioChannel.cs
using System;
using System.Collections.Generic;
using System.Text;
using urakawa.media;

namespace urakawa.property.channel
{
	/// <summary>
	/// A <see cref="Channel"/> that only accepts <see cref="IAudioMedia"/>
	/// </summary>
	public class AudioChannel : Channel
	{
		internal AudioChannel(ChannelsManager chMgr) : base(chMgr) {}

		/// <summary>
		/// Determines if a given <see cref="IMedia"/> can be accepted by the channel,
		/// which it can if it implements interface <see cref="IAudioMedia"/>
		/// </summary>
		/// <param name="m">The given media</param>
		/// <returns>A <see cref="bool"/> indicating if the given media can be accepted</returns>
		public override bool canAccept(urakawa.media.IMedia m)
		{
			if (!base.canAccept(m)) return false;
			if (m is IAudioMedia) return true;
            if (m is SequenceMedia)
            {
                foreach (IMedia sm in ((SequenceMedia)m).getListOfItems())
                {
                    if (!(sm is IAudioMedia)) return false;
                }
                return true;
            }
			return false;
		}
	}
}
